In [10]:
import pandas as pd
car = pd.read_excel('/content/Car data.xlsx')

In [11]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [4]:
car.shape

(432, 15)

# **1. Data Cleaning**

find all null values and if there are any replace with the mean of that column


In [12]:
car.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      2
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

In [14]:
car['Cylinders'].head()

0    6.0
1    4.0
2    4.0
3    6.0
4    6.0
Name: Cylinders, dtype: float64

In [27]:
car['Cylinders'].fillna(car['Cylinders'].mean(),inplace = True)

In [28]:
car['Cylinders'].isnull().sum()

0

2. Check what are the make are there in our dataset and what is the count (occurance) of each make in data set

In [32]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [31]:
car['Make'].value_counts()

Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              22
Audi             19
Honda            18
Nissan           17
Volkswagen       15
Chrysler         15
Dodge            13
Hyundai          13
Mitsubishi       13
Volvo            12
Jaguar           12
Subaru           11
Pontiac          11
Mazda            11
Lexus            11
Kia              11
Buick             9
Mercury           9
Lincoln           9
Saturn            8
Cadillac          8
Suzuki            8
Infiniti          8
GMC               8
Acura             7
Porsche           7
Saab              7
Land Rover        3
Oldsmobile        3
Jeep              3
Scion             2
Isuzu             2
MINI              2
Hummer            1
Name: Make, dtype: int64

3. Show all the records where origin is Asia or Europe

In [33]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [34]:
car[car['Origin'].isin(['Asia','Europe'])]

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0         197        21           28   
428  $40,083          2.3        5.0         242        20           26   
429  $42,573          2.9        6.0         268        19           26   
430  $24,641          1.9        4.0         170        22           29   
431  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
427    3450        105     186  
428    3450        105     186  
429    3653        110     190  
430    2822        101     180  
431    3823        109     186  

[285 rows x 15 columns]

4. Removing all the records where weight is above 4000

In [35]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [41]:
car[~(car['Weight']> 4000)]

Make                    Model   Type  Origin DriveTrain      MSRP  \
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
5    Acura  3.5 RL w/Navigation 4dr  Sedan    Asia      Front  $46,100    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
5    $41,100          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0         197        21           28   
428  $40,083          2.3        5.0         242        20           26   
429  $42,573          2.9        6.0         268        19           26   
430  $24,641          1.9        4.0         170        22           29   
431  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
5      3893        115     197  
..      ...        ...     ...  
427    3450        105     186  
428    3450        105     186  
429    3653        110     190  
430    2822        101     180  
431    3823        109     186  

[329 rows x 15 columns]

In [45]:
432-329

103

103 records are removed

5. Increase all the values of MPG_City columns by 3

In [46]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

In [48]:
car['MPG_City']=car['MPG_City'].apply(lambda x:x+3)

In [49]:
car.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        20           23    4451   
1         2.0        4.0         200        27           31    2778   
2         2.4        4.0         200        25           29    3230   
3         3.2        6.0         270        23           28    3575   
4         3.5        6.0         225        21           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197